contrib/lib/src/space_helmet/helmet.rs

```diff
@@ -0,0 +1,287 @@
+///  let rocket = rocket::ignite().attach(helmet);
+/// ```
+///
+/// `SpaceHelmet` supports the builder pattern to configure multiple policies
+/// ```rust
+/// # extern crate rocket;
+/// # extern crate rocket_contrib;
+///  use rocket_contrib::space_helmet::{HSTSPolicy, ExpectCTPolicy, ReferrerPolicy, SpaceHelmet};
+///  let helmet = SpaceHelmet::default()
+///     .hsts(HSTSPolicy::default())
+///     .expect_ct(ExpectCTPolicy::default())
+///     .referrer_policy(ReferrerPolicy::default());
+///
+///  let rocket = rocket::ignite().attach(helmet);
+/// ```
+///
+/// # TLS and HSTS
+///
+/// If tls is enabled when rocket
+/// [launched()'ed](../../rocket/fairing/trait.Fairing.html) in a
+/// non-development environment e.g.
+/// [Staging]( ../../rocket/struct.Config.html#method.staging)
+/// or [Production]( ../../rocket/struct.Config.html#method.production),
+/// `SpaceHelmet` enables hsts with its default policy and issue a warning.
+/// To get rid of this warning, set an [hsts](#method.hsts) policy if you are
+/// using tls.
+///
+pub struct SpaceHelmet<'a> {
+    expect_ct_policy: Option<ExpectCTPolicy<'a>>,
+    no_sniff_policy: Option<NoSniffPolicy>,
+    xss_protect_policy: Option<XSSPolicy<'a>>,
+    frameguard_policy: Option<FramePolicy<'a>>,
+    hsts_policy: Option<HSTSPolicy>,
+    force_hsts_policy: Option<HSTSPolicy>,
+    force_hsts: AtomicBool,
+    referrer_policy: Option<ReferrerPolicy>,
+}
+
+// helper for Helmet.apply
+macro_rules! try_apply_header {
+    ($self:ident, $response:ident, $policy_name:ident) => {
+        if let Some(ref policy) = $self.$policy_name {
+            if $response.set_header(policy) {
+                warn_!(
+                    "(Space Helmet): set_header failed, found existing header \"{}\"",
+                    Header::from(policy).name
+                );
+            }
+        }
+    };
+}
+
+impl<'a> SpaceHelmet<'a> {
+    /// Returns a new `SpaceHelmet` instance with `no_sniff`, `frameguard`,
+    /// and `xss_protect` enabled by default.
+    ///
+    /// # Example
+    ///
+    /// ```rust
+    /// # extern crate rocket;
+    /// # extern crate rocket_contrib;
+    /// use rocket_contrib::space_helmet::SpaceHelmet;
+    ///
+    /// # #[allow(unused_variables)]
+    /// let helmet = SpaceHelmet::default();
+    /// ```
+    pub fn default() -> Self {
+        Self {
+            expect_ct_policy: None,
+            no_sniff_policy: Some(NoSniffPolicy::default()),
+            frameguard_policy: Some(FramePolicy::default()),
+            xss_protect_policy: Some(XSSPolicy::default()),
+            hsts_policy: None,
+            force_hsts_policy: Some(HSTSPolicy::default()),
+            force_hsts: AtomicBool::new(false),
+            referrer_policy: None,
+        }
+    }
+
+    /// Same as `SpaceHelmet::default()`, see `SpaceHelmet::default()` for example.
+    pub fn new() -> Self {
+        SpaceHelmet::default()
+    }
+
+    /// Sets the [X-XSS-Protection](
```

Comment:
Ideally this would link to the section in the module/struct-level docs that describes this header. That section can then link to the Mozilla docs for further information.